packages/flutter_tools/bin/podhelper.rb

```diff
@@ -19,6 +19,29 @@ def flutter_ios_podfile_setup; end
 # Same as flutter_ios_podfile_setup for macOS.
 def flutter_macos_podfile_setup; end
 
+# Determine whether the target depends on Flutter (including transitive dependency)
+def is_depend_on_flutter(target, platform)
+  if platform == 'ios'
+    engine_pod_name = 'Flutter'
+  elsif platform == 'macos'
+    engine_pod_name = 'FlutterMacOS'
+  end
+
+  unless defined? engine_pod_name
+    raise 'Platform parameter can only be ios or macos.'
+  end
+
+  target.dependencies.any? do |dependency|
+    if dependency.name == engine_pod_name
+      return true
+    end
+
+    if is_depend_on_flutter(dependency.target, platform)
+      return true
+    end
+  end
```

Comment:
Please explicitly `return false` at the end.

Reply:
Wow, the stuck problem seems to be here, after adding `return false` at the end, the stuck disappeared